### PR TITLE
Fixes #961 - different abstraction for user message, take 1

### LINKED ISSUE #961: Different abstraction for user error/invalid messages?
Looking at `CashFlowRow` in src/forms/cashflow.js, the place 'invalid' message has ended up seems tangled unnecessarily (mostly my doing, I suspect). I'd like to look at other ways to abstract it that make more sense.

### PR DESCRIPTION
**Pull request by knod**
#961

### CODE REVIEW

File: src/forms/formHelpers.js [hunk 1]
```diff
@@ -1,33 +1,75 @@
...
+ * @param {string|object|null} [message] To be rendered above
+ *     the children.
+ * @param {boolean} [isUserError] Is this an error message?
+ *     Another option would be `type` instead - descriptor(s)
+ *     for the message that will currently be turned into
+ *     classes. Things like 'invalid'.
+ *
+ * @returns React element
+ */
+const WithMessageAbove = function ({ children, ariaName, isUserError, message, className }) {
```
Comment (pporfilio) [inline src/forms/formHelpers.js]: I like that you've added the ariaName to make it easy to associate with another node.

File: src/forms/formHelpers.js [hunk 2]
```diff
@@ -1,33 +1,75 @@
...
-    result = (
+  let messageElem = null;
+  if (message) {
+
+    let messageClasses = `message-above `;
+    if (isUserError) {
+      messageClasses += `invalid`;
+    }
+    // `id` is for aria description hook
+    // tabIndex is for
```
Comment (pporfilio) [inline src/forms/formHelpers.js]: I think you meant to add a note about internet explorer here.

**Review by pporfilio — APPROVED** [review]

File: src/forms/formHelpers.js (hunk 2 above)
Comment (knod) [reply src/forms/formHelpers.js]: lol, thanks for catching that

**Comment by turnerhayes** [comment]
Looks good, but I wanted to put a discussion question out there; is this premature abstraction? It seems it's only currently being used in one file; are we sure this is something that's going to be reused?

**Review by turnerhayes — APPROVED** [review]

**Comment by knod** [comment]
We do plan to re-use it as we move forward. Also, It seemed like a more concrete abstraction to me. We also discussed that it could use a better name.

**Pull request merged**